client/components/jetpack/threat-history-list/index.tsx

```diff
@@ -1,217 +1,192 @@
-import SimplifiedSegmentedControl from 'calypso/components/segmented-control/simplified';
+import { useTranslate } from 'i18n-calypso';
+import { useMobileBreakpoint } from '@automattic/viewport-react';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
-import { getSelectedSiteSlug, getSelectedSite } from 'calypso/state/ui/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import isRequestingJetpackScanThreatCounts from 'calypso/state/selectors/is-requesting-jetpack-scan-threat-counts';
 import isRequestingJetpackScanHistory from 'calypso/state/selectors/is-requesting-jetpack-scan-history';
-import getSiteScanHistory from 'calypso/state/selectors/get-site-scan-history';
-import contactSupportUrl from 'calypso/lib/jetpack/contact-support-url';
-import { withLocalizedMoment } from 'calypso/components/localized-moment';
-import { useThreats } from 'calypso/lib/jetpack/use-threats';
-import { Threat } from 'calypso/components/jetpack/threat-item/types';
-import { Site } from 'calypso/my-sites/scan/types';
-
-/**
- * Style dependencies
- */
-import './style.scss';
-
-type FilterValue = 'all' | 'fixed' | 'ignored';
-type FilterOption =
-	| {
-			value: 'all';
-			label: 'All';
-	  }
-	| {
-			value: 'fixed';
-			label: 'Fixed';
-	  }
-	| {
-			value: 'ignored';
-			label: 'Ignored';
-	  };
-
-const filterOptions: FilterOption[] = [
-	{ value: 'all', label: 'All' },
-	{ value: 'fixed', label: 'Fixed' },
-	{ value: 'ignored', label: 'Ignored' },
-];
-
-interface ThreatStatusFilterProps {
-	isPlaceholder: boolean;
-	onSelect: ( ...args: any[] ) => void;
-	initialSelected: FilterValue;
-}
-
-const ThreatStatusFilter: React.FC< ThreatStatusFilterProps > = ( {
-	isPlaceholder,
-	onSelect,
-	initialSelected,
-} ) => {
-	return isPlaceholder ? (
-		<div className="threat-history-list__filters is-placeholder"></div>
-	) : (
-		<SimplifiedSegmentedControl
-			className="threat-history-list__filters"
-			options={ filterOptions }
-			onSelect={ onSelect }
-			initialSelected={ initialSelected }
-			primary
-		/>
-	);
+import getScanSiteThreatCounts from 'calypso/state/selectors/get-site-scan-threat-counts';
+import getScanSiteHistory from 'calypso/state/selectors/get-site-scan-history';
+import QueryJetpackScanThreatCounts from 'calypso/components/data/query-jetpack-scan-threat-counts';
+import QueryJetpackScanHistory from 'calypso/components/data/query-jetpack-scan-history';
+import Pagination from 'calypso/components/pagination';
+import ThreatStatusFilter, { FilterValue, FilterOption } from './threat-status-filter';
+import ListItems, { ListItemsPlaceholder } from './list-items';
+
+const THREATS_PER_PAGE = 10;
+
+const trackFilterChange = ( siteId: number, filter: string ) =>
+	recordTracksEvent( 'calypso_jetpack_scan_history_filter', {
+		site_id: siteId,
+		filter,
+	} );
+
+const getFilteredThreatCount = (
+	threatCounts: { [ key: string ]: number },
```

Comment:
Non-blocking comment: for documentation purposes, I think we should write a type for this since it's the response object we get from the endpoint and we won't know what shape it has unless we check the request response.